### PR TITLE
fix(symbolication): Correct aliased source names

### PR DESCRIPTION
**Pull request by flub**
When we had aliased sources the source names were added before the
aliasing was undone.  This resulted in some random name from before
the alias showing up.

This instead consolidates all the candidates post-processing to one
location and makes sure the alias reversing happens before the names
are added as well as that aliased source have their name present.

### CODE REVIEW

File: src/sentry/lang/native/symbolicator.py [hunk 1]
```diff
@@ -193,114 +191,6 @@ def process_payload(self, stacktraces, modules, signal=None):
         )
 
 
-def reverse_source_aliases(response, builtin_sources=None):
```
Comment (flub) [inline src/sentry/lang/native/symbolicator.py:196]: this has been merged into `SymbolicatorSession._process_response`

File: src/sentry/lang/native/symbolicator.py [hunk 2]
```diff
@@ -193,114 +191,6 @@ def process_payload(self, stacktraces, modules, signal=None):
...
-    reverse_aliases = reverse_aliases_map(builtin_sources)
-
-    for module in response.get("modules", []):
-        for candidate in module.get("candidates", []):
-            source_id = candidate["source"]
-            if source_id in reverse_aliases:
-                candidate["source"] = reverse_aliases.get(source_id)
-
-
-def reverse_aliases_map(builtin_sources):
```
Comment (flub) [inline src/sentry/lang/native/symbolicator.py:214]: This has been moved verbatim to low lower in the file.

File: src/sentry/lang/native/symbolicator.py [hunk 3]
```diff
@@ -193,114 +191,6 @@ def process_payload(self, stacktraces, modules, signal=None):
...
-            try:
-                aliased_source = builtin_sources[aliased_source]
-                aliased_id = aliased_source["id"]
-            except KeyError:
-                continue
-            reverse_aliases[aliased_id] = self_id
-    return reverse_aliases
-
-
-def redact_internal_sources(response):
```
Comment (flub) [inline src/sentry/lang/native/symbolicator.py:237]: This has been moved verbatim to low lower in the file.

File: src/sentry/lang/native/symbolicator.py [hunk 4]
```diff
@@ -193,114 +191,6 @@ def process_payload(self, stacktraces, modules, signal=None):
...
-    not leak any internal details.
-
-    Note that this modifies the argument passed in, thus redacting in-place.  It still
-    returns the modified response.
-    """
-    for module in response.get("modules", []):
-        redact_internal_sources_from_module(module)
-
-
-def redact_internal_sources_from_module(module):
```
Comment (flub) [inline src/sentry/lang/native/symbolicator.py:252]: This has been moved verbatim to low lower in the file.

File: src/sentry/lang/native/symbolicator.py (hunk 2 above)
Comment (jan-auer) [reply src/sentry/lang/native/symbolicator.py:214]: What was the reason for moving this? It is now declared after use.

**Review by jan-auer — APPROVED** [review]
Looks good to me. 

The moves create an unnecessarily large diff, could we consider reverting those if they don't serve a purpose?

File: src/sentry/lang/native/symbolicator.py (hunk 2 above)
Comment (flub) [reply src/sentry/lang/native/symbolicator.py:214]: I did this in an attempt to follow file's current style.  `get_options_for_project()` and `get_sources_for_project()` are used in `Symbolicator` and declared after it.  Hence these functions are used in `SymbolicatorSession` and declared after it.

**Comment by flub** [comment]
> The moves create an unnecessarily large diff, could we consider reverting those if they don't serve a purpose?

I'd prefer to keep the moves as they keep the current file's structure.  Or at least that's what it looks like to me.  Let me know if you disagree and consider this high priority though.

File: src/sentry/lang/native/symbolicator.py (hunk 2 above)
Comment (jan-auer) [reply src/sentry/lang/native/symbolicator.py:214]: Thanks for explaining, makes sense 👍

**Comment by jan-auer** [comment]
Thanks, no need to revert the moves then.

**Pull request merged**